array-methods.js

```diff
@@ -58,20 +87,62 @@ var stateSums = null;
     round this number to the nearest dollar before moving on.
   )
  */
-var sumOfHighInterests = null;
+var sumOfHighInterests = Object.entries(stateSums)
+  .filter(function(element) {
+    //element[0] targets the key which is the state
+    //ex [AL,2839171]
+    if (!getStates.includes(element[0])) {
+      return true;
+    }
+  })
+  .map(function(element) {
+    //element[1] targets the amount
+    return Math.round(Number(element[1] * 0.189));
+  })
+  .filter(function(element) {
+    if (element > 50000) {
+      return true;
+    }
+  })
+  .reduce(function(acc, current) {
+    return acc + current;
+  });
 
 /*
   set `lowerSumStates` to be an array of two letter state
   abbreviations of each state where the sum of amounts
   in the state is less than 1,000,000
  */
-var lowerSumStates = null;
+var lowerSumStates = Object.entries(stateSums)
+  .filter(function(element) {
+    if (element[1] < 1000000) {
+      return true;
+    }
+  })
+  .reduce(function(acc, current) {
+    //why does this only work if i return acc AFTER pushing into the empty array
+    acc.push(current[0]);
```

Comment:
if you put a return here, then you're returning the result of the `.push` method which is undefined.